openedx/core/djangoapps/programs/tasks/v1/tasks.py

```diff
@@ -194,8 +194,14 @@ def award_program_certificates(self, username):
                 )
             except Exception:  # pylint: disable=broad-except
                 # keep trying to award other certs, but retry the whole task to fix any missing entries
-                LOGGER.exception('Failed to award certificate for program %s to user %s', program_uuid, username)
-                retry = True
+                warning_message = 'Failed to award certificate for program {uuid} to user {username}.'.format(
```

Comment:
I think it would be best if you add retrying at the end as well `Failed to award certificate for program {uuid} to user {username}, retrying.` It will add more clarity about this warning.

Reply:
we have a info message below at [this line](https://github.com/edx/edx-platform/pull/16822/files#diff-e88c124a2f70ddddd3e71c6b41fc95d9L202) for retrying.